src/views/home/Index.svelte

```diff
@@ -5,8 +5,52 @@
 
   import Link from "@app/components/Link.svelte";
   import ProjectCard from "@app/components/ProjectCard.svelte";
+  import { api, httpdStore } from "@app/lib/httpd";
+  import { loadProjectActivity } from "@app/lib/commit";
 
   export let projects: ProjectBaseUrlActivity[];
+
+  const networkHeading =
+    'Explore <span class="txt-bold">projects</span> the Radicle network.';
+  const localHeading =
+    'Explore <span class="txt-bold">projects</span> on your local node.';
+
+  $: if ($httpdStore.state !== "stopped") {
+    fetchLocalProjects()
+      .then(localProjects => {
+        switchProjects(localHeading, localProjects);
+      })
+      .catch(() => {
+        switchProjects(networkHeading, projects);
+      });
+  } else {
+    switchProjects(networkHeading, projects);
+  }
```

Comment:
This code introduces a couple problems:
1. We now have two places where project data is fetched: here and in `home/router.ts`.
2. The is susceptible to a race condition: If `$httpdStore` switches quickly, two `fetchLocalProjects()` tasks may run concurrently and call `switchProjects` out of order.
3. If there is a local node running, the user sees the pinned projects first and then the view updates to the local projects.

An alternative implementation would be to move the fetching of local projects into `loadHomeRoute()` in `home/router.ts`. What we loose then is some reactivity. When you change the port the page is not automatically updated. But I think that’s actually good because there is no loading indicator anyways and you want to avoid a situation where part of the page is updated without the  user knowing about it.

Reply:
Ok yeah the loading of the projects in the router was my first version, maybe once we have some notification system we can show something that says "We detected a running local node, do you want to reload to see your local projects?"

Reply:
Addressed in 1f9e2a4